### PR TITLE
WIP docs-requirements.txt: Upgrade sphinx to 1.5.5 and add sphinx_rtd_theme

### PR DESCRIPTION
**Pull request by yash-nisar**
Closes https://github.com/coala/coala-bears/issues/1554

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # NOTE: This file is parsed by .ci/generate_bear_requirements.py
 # Use >= for development versions so that source builds always work
-coala>=0.11.0.dev20170322155718
+coala==0.11.0.dev20170404142421
```
Comment (supergr35) [inline requirements.txt]: unrelated change, or required?

Comment (yash-nisar) [inline requirements.txt]: Unrelated, was investigating about the bug. :)

Comment (supergr35) [inline requirements.txt]: In that case does it belong in this PR?

Comment (yash-nisar) [inline requirements.txt]: Nope, will change it.

**Comment by userzimmermann** [comment]
@yash-nisar Very strange! Locally it works for me using your branch:

```
Running Sphinx v1.5.5
loading pickled environment... failed: build environment version not current
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 46 source files that are out of date
updating environment: 46 added, 0 changed, 0 removed
WARNING:root:StyleLintBear: Using unnecessary capturing groups affects the performance of coala. You should use '(?:<pattern>)' instead of '(<pattern>)' for your regex.
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex py-modindex
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded.

Build finished. The HTML pages are in _build/html.
make: Leaving directory '/c/Users/Zimmermann/Projects/coala/bears/docs'
```

**Comment by yash-nisar** [comment]
Whoops ! We seem to be stuck here.

**Comment by jayvdb** [comment]
Why is this adding sphinx_rtd_theme ?
And please see https://github.com/coala/coala-bears/pull/1819

File: docs-requirements.txt [hunk 2]
```diff
@@ -1 +1,3 @@
-sphinx==1.3.5
+sphinx~=1.5.5
+sphinx-argparse~=0.1.15
```
Comment (jayvdb) [inline docs-requirements.txt]: Was there a reason you added this?

Comment (yash-nisar) [inline docs-requirements.txt]: Nope, just for testing purposes.

**Comment by yash-nisar** [comment]
Isn't @userzimmermann doing this issue ? or are we planning to include my commit below his commit ?

**Comment by jayvdb** [comment]
@yash-nisar , I am trying to make sure I understand what you discovered during this patch .  It may be relevant to the other patch.

**Comment by userzimmermann** [comment]
@yash-nisar I can rebase my stuff from https://github.com/coala/coala-bears/pull/1819 on your commit if you like... ;)

**Comment by userzimmermann** [comment]
@yash-nisar And please take a look at the discussion in https://github.com/coala/coala-bears/pull/1819#pullrequestreview-42875060

Maybe you have some further idea or advice on this :)

**Comment by yash-nisar** [comment]
Closing, since work is being continued in https://github.com/coala/coala-bears/pull/1819

**Comment by userzimmermann** [comment]
I took over https://github.com/coala/coala-bears/issues/1554 in https://github.com/coala/coala-bears/pull/1819
